.eleventy.js

```diff
@@ -1,4 +1,28 @@
 const EleventyFetch = require("@11ty/eleventy-fetch");
+const { DOMParser, parseHTML } = require('linkedom');
+const probe = require('probe-image-size');
+
+const TITLE_SELECTORS = [
+  { selector: `meta[property='og:title']`, attribute: 'content' },
+  { selector: `meta[property='twitter:title']`, attribute: 'content' },
+  { selector: 'title', attribute: 'textContent'}
+]
+
+const DESC_SELECTORS = [
+  { selector: `meta[property='description']`, attribute: 'content' },
+  { selector: `meta[property='og:description']`, attribute: 'content' },
+  { selector: `meta[property='twitter:description']`, attribute: 'content' },
+]
+
+const IMAGE_SELECTORS = [
+  { selector: `meta[property='og:image']`, attribute: 'content' },
+  { selector: `meta[property='twitter:image']`, attribute: 'content' },
+]
+
+const LOGO_SELECTORS = [
+  { selector: `link[rel='apple-touch-icon']`, attribute: 'href' },
+  { selector: `link[rel='icon']`, attribute: 'href' },
+]
```

Comment:
Might want to reorganise these arrays as from reading them it appears that the first instance of each will get returned. The globally accepted ones should come first followed by the platform specific ones, like twitter and apple icon.